cclient/serialization/json/json_serializer.c

```diff
@@ -8,13 +8,113 @@ void init_json_serializer(serializer_t* serializer) {
   serializer->vtable = json_vtable;
 }
 
-void json_find_transactions_serialize_request(
+retcode_t json_array_to_utarray(cJSON* obj, const char* obj_name,
+                                UT_array* ut) {
+  cJSON* json_item = cJSON_GetObjectItemCaseSensitive(obj, obj_name);
+  if (cJSON_IsArray(json_item)) {
+    cJSON* current_obj = NULL;
+    cJSON_ArrayForEach(current_obj, json_item) {
+      if (current_obj->valuestring != NULL) {
+        utarray_push_back(ut, &current_obj->valuestring);
+      }
+    }
+  } else {
+    cJSON_Delete(obj);
+    return RC_CCLIENT_JSON_PARSE;
+  }
+  return RC_OK;
+}
+
+retcode_t json_find_transactions_serialize_request(
     const serializer_t* const s, const find_transactions_req_t* const obj,
-    char* out) {}
+    char_buffer* out) {
+  cJSON* json_root = cJSON_CreateObject();
+  if (json_root == NULL) return RC_CCLIENT_JSON_CREATE;
+
+  cJSON_AddItemToObject(json_root, "command",
+                        cJSON_CreateString("findTransactions"));
+
+  if (utarray_len(obj->addresses) > 0) {
+    cJSON* addrs = cJSON_CreateArray();
+    if (addrs == NULL) goto err;
+
+    cJSON_AddItemToObject(json_root, "addresses", addrs);
+    char** p = NULL;
+    while ((p = (char**)utarray_next(obj->addresses, p))) {
+      cJSON_AddItemToArray(addrs, cJSON_CreateString(*p));
+    }
+  }
+
+  if (utarray_len(obj->approvees) > 0) {
+    cJSON* approvees = cJSON_CreateArray();
+    if (approvees == NULL) goto err;
+
+    cJSON_AddItemToObject(json_root, "approvees", approvees);
+    char** p = NULL;
+    while ((p = (char**)utarray_next(obj->approvees, p))) {
+      cJSON_AddItemToArray(approvees, cJSON_CreateString(*p));
+    }
+  }
+
+  if (utarray_len(obj->bundles) > 0) {
+    cJSON* bundles = cJSON_CreateArray();
+    if (bundles == NULL) goto err;
+
+    cJSON_AddItemToObject(json_root, "bundles", bundles);
+    char** p = NULL;
+    while ((p = (char**)utarray_next(obj->bundles, p))) {
+      cJSON_AddItemToArray(bundles, cJSON_CreateString(*p));
+    }
+  }
+
+  if (utarray_len(obj->tags) > 0) {
```

Comment:
You should define a function that take `utarray` , json_root and label and return `cJSONArray` to avoid duplications

Reply:
sure